### PR TITLE
added variables for pufferfish, also added setup only test

### PR DESCRIPTION
**Pull request by shotah**
Fixing pufferfish bug, as well as adding a test for it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -539,7 +539,7 @@ A [Pufferfish](https://github.com/pufferfish-gg/Pufferfish) server, which is "a
 
     -e TYPE=PUFFERFISH
 
-> NOTE: The `VERSION` variable is used to select a Pufferfish branch to download from. The available options are "LATEST" and "1.18"
+> NOTE: The `VERSION` variable is used to select a Pufferfish branch to download from. The available options are "LATEST"
```
Comment (itzg) [inline README.md:542]: That last sentence is dangling. I'm thinking the whole note can just become "VERSION can only be set to LATEST, which is the default."

**Review by itzg — CHANGES_REQUESTED** [review]
Thanks! Yeah, that R0.1 qualifier is awkward but at least it'll get caught during tests if they ever change it.

File: scripts/start-deployPufferfish [hunk 2]
```diff
@@ -1,44 +1,43 @@
...
 
-log "Using Pufferfish-${PUFFERFISH_BRANCH} branch"
+PUFFERFISH_BUILD_JSON=$(curl -X GET -s "https://ci.pufferfish.host/job/Pufferfish-${VANILLA_VERSION%.*}/${PUFFERFISH_BUILD}/api/json")
+# Example: "url": "https://ci.pufferfish.host/job/Pufferfish-1.18/50/",
+PUFFERFISH_BUILD_URL=$(jq -n "$PUFFERFISH_BUILD_JSON" | jq -jc '.url // empty' )
+# Example: "fileName": "pufferfish-paperclip-1.18.2-R0.1-SNAPSHOT-reobf.jar",
+PUFFERFISH_BUILD_FILENAME=$(jq -n "$PUFFERFISH_BUILD_JSON" | jq -jc '.artifacts[].fileName // empty' )
+PUFFERFISH_BUILD_DOWNLOAD_URL="${PUFFERFISH_BUILD_URL}artifact/build/libs/${PUFFERFISH_BUILD_FILENAME}"
 
-export SERVER=pufferfish-${PUFFERFISH_BRANCH}-${PUFFERFISH_BUILD}.jar
```
Comment (itzg) [inline scripts/start-deployPufferfish:23]: SERVER is a very important variable used in the downstream scripts. Is the server starting up with this removed?

File: scripts/start-deployPufferfish [hunk 3]
```diff
@@ -1,44 +1,43 @@
...
 
 log "Removing old Pufferfish versions ..."
 shopt -s nullglob
 for f in pufferfish-*.jar; do
-  [[ $f != $SERVER ]] && rm $f
+  [[ $f != "$PUFFERFISH_BUILD_FILENAME" ]] && rm "$f"
```
Comment (itzg) [inline scripts/start-deployPufferfish:29]: ...why the change from SERVER in the first place?

File: scripts/start-deployPufferfish (hunk 2 above)
Comment (shotah) [reply scripts/start-deployPufferfish:23]: using setup only for testing. Let me set file name to server.

File: scripts/start-deployPufferfish (hunk 3 above)
Comment (shotah) [reply scripts/start-deployPufferfish:29]: because  I was getting away from the static concatenation of filename. Will pivot file name to SERVER.

**Review by itzg — APPROVED** [review]
🐡!
